### DISCUSSION TITLE
How to ignore filenames

### DISCUSSION THREAD

**Discussion by josh-gree** [whitews/FlowKit#217] [Q&A] [answered]
Hi - thanks for creating this amazing tool!

I would love to understand if its possible to completely ignore anything about sample filenames and ids and the like - I just want to be able to take the gating tree from some wsp and apply it arbitrarily to a given sample - it may well be the case that the gating tree and the sample don't work together but I would be validating that separately and externally from flowkit - I am sure the machinery exists to do this within FlowKit but not quite sure how to get to it?

**Comment by whitews** [comment, marked as answer]
Hi Josh,

Great to hear you find the library useful. 

Yes, there is a way to do this. Once you load a FlowJo .wsp file as `Workspace` instance, use the [`get_gating_strategy`](https://flowkit.readthedocs.io/en/latest/workspace.html#flowkit.Workspace.get_gating_strategy) method to retrieve a `GatingStrategy` instance. That method takes a sample_id argument to get the specific gating strategy for that sample. Note, if you don't have the FCS sample, you can load the .wsp file using `ignore_fcs_files=True` and the workspace will still load the gating information so you can retrieve it. Then, the `GatingStrategy` instance can be used against any sample using its [`gate_sample`](https://flowkit.readthedocs.io/en/latest/gating_strategy.html#flowkit.GatingStrategy.gate_sample) method.

The `GatingStrategy` class is the engine behind both the `Session` and `Workspace` classes. However, it was also made to be used for such a scenario where you don't want the library to manage samples for you.

Hope this helps and let me know if that's what you were looking for.

Kind regards,
Scott
